src/plugins/profiling/server/routes/search_flamechart.ts

```diff
@@ -189,19 +192,52 @@ async function queryFlameGraph(
   const executableDocIDs = new Set<string>(); // Set of unique executable FileIDs.
   const stackTraceIDs = [...stackTraceEvents.keys()];
   const chunkSize = Math.floor(stackTraceEvents.size / nQueries);
-  const chunks = chunk(stackTraceIDs, chunkSize);
+  let chunks = chunk(stackTraceIDs, chunkSize);
+
+  if (chunks.length !== nQueries) {
+    // The last array element contains the remainder, just drop it as irrelevant.
+    chunks = chunks.slice(0, nQueries);
+  }
 
   const stackResponses = await logExecutionLatency(
     logger,
-    'mget query (' + nQueries + ' parallel) for ' + stackTraceEvents.size + ' stacktraces',
+    (testing ? 'search' : 'mget') +
+      ' query (' +
+      nQueries +
+      ' parallel) for ' +
+      stackTraceEvents.size +
+      ' stacktraces',
```

Comment:
Probably more readable with a `String.format()` call?

Reply:
Yeah, we can fix this in case we keep the search (which I don't think we will), together with a bunch of other things.